pgmpy/factors/continuous/ContinuousFactor.py

```diff
@@ -0,0 +1,259 @@
+import six
+import numpy as np
+import scipy.integrate as integrate
+
+
+class ContinuousFactor(object):
+    """
+    Base class for factors representing various multivariate
+    representations.
+    """
+    def __init__(self, variables, pdf):
+        """
+        Parameters
+        ----------
+        variables: list or array-like
+            The variables for wich the distribution is defined.
+
+        pdf: function
+            The probability density function of the distribution.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy.special import beta
+        # Two variable drichlet ditribution with alpha = (1,2)
+        >>> def drichlet_pdf(x, y):
+        ...     return (np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> drichlet_factor = ContinuousFactor(['x', 'y'], drichlet_pdf)
+        >>> drichlet_factor.scope()
+        ['x', 'y']
+        >>> drichlet_factor.assignemnt(5,6)
+        226800.0
+        """
+        if not isinstance(variables, list):
```

Comment:
@yashu-seth We just want to accept something that is iterable like set, dict, list, tuple, np.array. User can pass anything, so you need to check if it's iterable or not. And because strings are iterables and we don't want to accept that we need to check for that also.